src/index.js

```diff
@@ -5,11 +5,20 @@ import {
   curdFunctionality,
   retrieveTodoListFromLocalStorage,
 } from './curd.js';
+import { clearCompleted } from './status.js';
 
 document.addEventListener('DOMContentLoaded', () => {
+  curdFunctionality();
+
   const storedTodoList = retrieveTodoListFromLocalStorage();
   todoList.push(...storedTodoList);
 
-  curdFunctionality();
+  const clearButton = document.querySelector('.clearAll');
+  clearButton.addEventListener('click', (e) => {
+    e.preventDefault();
+    clearCompleted(todoList);
+  });
+  populateList();
```

Comment:
@umarahgul Great work  implementing the clearCompleted functionality, when I click the checkbox to update the status to completed and refresh the page tasks completed are automatically deleted, which shouldn't be the case, the tasks should be deleted only when the use clicks on the  "Clear all completed" button.

Reply:
@danmainah . Thanks for revieweing my code and showing appreciation. I got your point and I didnt notice it earlier. However, I need to know that once completed=true and i do clear-all-completed; to my understanding , the deleted tasks shall not not show in the local storage with completed=true, because they have been deleted using clear-all-completed
  waiting for your response, before I proceed further. Thank you once again.

Reply:
@umarahgul The task can only be removed by clicking on the "Clear all completed" button. A user can change the completed status without clicking the "Clear all completed" button , in that case the change in status will be stored in the local storage incase the user will want to remove the task or keep tab of completed tasks.